### PR TITLE
CB-21417 hadoop-aws native openssl mode breaks on DE datahub in govcloud-int

### PR DESCRIPTION
**Pull request by adarshsharma1983**
Disable openssl for AWS GOV also

See detailed description in the commit message.

### CODE REVIEW

File: template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java [hunk 1]
```diff
@@ -290,6 +299,11 @@ public Builder withCloudPlatform(CloudPlatform cloudPlatform) {
             return this;
         }
 
+        public Builder withPlatformVariant(String platformVariant) {
```
Comment (doktoric) [inline template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java:302]: GeneralClusterConfigs already contains this

Comment (adarshsharma1983) [reply template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java:302]: I already checked that but found that it is being set with variant column in cluster table. I checked the same and I can see only null,CLOUDERA_MANAGER as distinct values for variant column in cluster table. thats why I did not use it.

public GeneralClusterConfigs generalClusterConfigs(StackDtoDelegate stack, Credential credential) {
        ClusterView cluster = stack.getCluster();
        boolean gatewayInstanceMetadataPresented = false;
        boolean instanceMetadataPresented = false;
        if (stack.getInstanceGroupDtos() != null && !stack.getInstanceGroupDtos().isEmpty()) {
            List<InstanceMetadataView> gatewayInstanceMetadata = stack.getNotTerminatedAndNotZombieGatewayInstanceMetadata();
            gatewayInstanceMetadataPresented = !gatewayInstanceMetadata.isEmpty() && stack.hasGateway();
            instanceMetadataPresented = true;
        }
        GeneralClusterConfigs generalClusterConfigs = new GeneralClusterConfigs();
        generalClusterConfigs.setClusterManagerIp(cluster.getClusterManagerIp());
        generalClusterConfigs.setInstanceGroupsPresented(instanceMetadataPresented);
        generalClusterConfigs.setGatewayInstanceMetadataPresented(gatewayInstanceMetadataPresented);
        generalClusterConfigs.setClusterName(cluster.getName());
        generalClusterConfigs.setStackName(stack.getName());
        generalClusterConfigs.setUuid(stack.getUuid());
        generalClusterConfigs.setUserName(cluster.getUserName());
        generalClusterConfigs.setPassword(cluster.getPassword());
        generalClusterConfigs.setCloudbreakAmbariUser(cluster.getCloudbreakAmbariUser());
        generalClusterConfigs.setCloudbreakAmbariPassword(cluster.getCloudbreakAmbariPassword());
        generalClusterConfigs.setNodeCount(stack.getFullNodeCount().intValue());
        generalClusterConfigs.setPrimaryGatewayInstanceDiscoveryFQDN(Optional.ofNullable(stack.getPrimaryGatewayInstance().getDiscoveryFQDN()));
        **generalClusterConfigs.setVariant(cluster.getVariant());**
        generalClusterConfigs.setAutoTlsEnabled(cluster.getAutoTlsEnabled());
        boolean userFacingCertHasBeenGenerated = StringUtils.isNotEmpty(stack.getSecurityConfig().getUserFacingKey())
                && StringUtils.isNotEmpty(stack.getSecurityConfig().getUserFacingCert());
        generalClusterConfigs.setKnoxUserFacingCertConfigured(userFacingCertHasBeenGenerated);
        generalClusterConfigs.setExternalFQDN(cluster.getFqdn());
        generalClusterConfigs.setEnableRangerRaz(cluster.isRangerRazEnabled());
        generalClusterConfigs.setGovCloud(credential.isGovCloud());
        generalClusterConfigs.setCreatorWorkloadUserCrn(stack.getCreator().getUserCrn());
        return generalClusterConfigs;
    }

Comment (doktoric) [reply template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java]: interesting.. @adarshsharma1983 can we open this pr against 2.70.0?

**Comment by lajosrodek** [comment]
Closing this in favor of #14667 (2.70) and #14669 (master / 2.71).

File: template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java (hunk 1 above)
Comment (lajosrodek) [reply template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java]: I will directly fix this in the 2.70 commit, but let me add this here nevertheless: The original approach would have the preferred way. `generalClusterConfigs.variant` has a different purpose, it is the cluster manager variant, not the cloud platform variant. As you also noted, `generalClusterConfigs.variant` has already been used in `BlueprintUpdaterConnectors.getBlueprintText()` to retrieve the appropriate `BlueprintUpdater` instance. It is, therefore, not appropriate to "repurpose" the already existing `generalClusterConfigs.variant` for a totally different kind of value in in one place, where it is still being used for something else in another place.

In short, we definitely need a dedicated `platformVariant` in `TemplatePreparationObject`. :smile:
